python/pyspark/sql/connect/session.py

```diff
@@ -418,6 +425,28 @@ def createDataFrame(
             # If no schema supplied by user then get the names of columns only
             if schema is None:
                 _cols = [str(x) if not isinstance(x, str) else x for x in data.columns]
+                infer_pandas_dict_as_map = (
+                    str(self.conf.get("spark.sql.execution.pandas.inferPandasDictAsMap")).lower()
+                    == "true"
+                )
+                if infer_pandas_dict_as_map:
+                    struct = StructType()
+                    pa_schema = pa.Schema.from_pandas(data)
+                    spark_type: Union[MapType, DataType]
+                    for field in pa_schema:
+                        field_type = field.type
+                        if isinstance(field_type, pa.StructType):
```

Comment:
How does this work for nested type cases?

Reply:
What does nested type cases mean you are concerned about? Maybe do you mean a `dict` within a `dict`, or a `list` within a `dict` for example? If so, the result depends entirely on Arrow's inferring logic, so it is difficult to give an exact answer for all cases. If you have a specific case that you're concerned about, let me add some testing to make sure it works.

Reply:
At least it will provide the same result for connect and non-connect for sure.